### PR TITLE
perlapi: Combine sv_unref, _flags

### PR DESCRIPTION
**Pull request by khwilliamson**

### CODE REVIEW

**Comment by jkeenan** [comment]
When I build this branch, call `perldoc pod/perlapi.pod` and search for `sv_unref_flags`, I come to:
```
    "sv_unref"
    "sv_unref_flags"
        These each unset the RV status of the SV, and decrement the
        reference count of whatever was being referenced by the RV. This can
        almost be thought of as a reversal of "newSVrv".

        The "flags" argument in "sv_unref_flags" can contain
        "SV_IMMEDIATE_UNREF" to force the reference count to be decremented
        (otherwise the decrementing is conditional on the reference count
        being different from one or the reference being a readonly SV).

        "sv_unref" merely calls "sv_unref_flags" with "flags" set to zero.

        See "SvROK_off".

            void  sv_unref      (SV *sv)
            void  sv_unref_flags(SV * const ref, const U32 flags)
```
Other things being equal, I would have expected the display of the two function calls -- the last two lines above -- to appear *before* the paragraph which begins, `The "flags" argument in "sv_unref_flags" can contain...`.  That way, the reader has already been prompted to note the difference between the two function calls and is mentally set up to be provided with an explanation of that difference.

Is this possible with the program we use to generate `pod/perlapi.pod`?  Would that be a better approach?

**Comment by khwilliamson** [comment]
That's an interesting idea.  I could see adding an apidoc-like command that says insert the prototypes "here" instead of the normal place, at the end of the entry.  But the normal place would have to remain the end of the entry, or perhaps replace the list at the top with the full signatures.  I will try experimenting with that and if I can get it to look ok to me, will post a PR to that effect.

Even if we did do that, I'm thinking this would be better to say what is in the new version I just pushed

File: sv.c [hunk 1]
```diff
@@ -10936,14 +10936,22 @@ S_sv_unglob(pTHX_ SV *const sv, U32 flags)
...
-(otherwise the decrementing is conditional on the reference count being
-different from one or the reference being a readonly SV).
+as a reversal of C<L</newSVrv>>.
+
+C<sv_unref_flags> has an extra parameter, C<flags>, which can contain
+the C<SV_IMMEDIATE_UNREF> bit to force the reference count to be decremented 
+no matter what.
+
+When that bit isn't set, or with plain C<sv_unref> always, the reference count
+will not be decremented if the referent is read-only, or if the count is 1.
```
Comment (tonycoz) [inline sv.c]: a) there's no read-only check, as the comment in the code indicates.  It looks like the SvREADONLY() check was removed in 04ca4930675 but the documentation wasn't updated.

b) even if the target SV has a single reference, and the UNREF flags is unset, the target SV is still mortalized and will be freed "later"

Comment (khwilliamson) [reply sv.c]: see the new wording

**Review by tonycoz — APPROVED** [review]

**Pull request merged**